src/Sylius/Bundle/CoreBundle/Doctrine/ORM/OrderRepository.php

```diff
@@ -164,8 +164,14 @@ public function createCheckoutsPaginator(array $criteria = null, array $sorting
      */
     public function countByCustomer(CustomerInterface $customer)
     {
-        $count = $this->createByCustomerQueryBuilder($customer)
+        $queryBuilder = $this->createQueryBuilder('o');
+
+        $count = $queryBuilder
             ->select('count(o.id)')
+            ->andWhere('o.customer = :customer')
+            ->andWhere($queryBuilder->expr()->notIn('o.state', ':states'))
+            ->setParameter('customer', $customer)
+            ->setParameter('states', [OrderInterface::STATE_CART, OrderInterface::STATE_CANCELLED])
```

Comment:
That's how we end up having `createQueryBuilderForEnabledByTaxonIdAndChannelAndLocale` 😄 

I agree that current method name is too generic, but let's discuss the names on Monday and standardize them all at once, because there are many more methods that needs renaming.